EEJT-0002/src/main/java/com/epam/esm/utils/validation/DataValidation.java

```diff
@@ -0,0 +1,90 @@
+                giftCertificate.getDuration() != null && giftCertificate.getDuration() >= 0 &&
+                isStringValid(giftCertificate.getDescription()) &&
+                giftCertificate.getPrice() != null &&
+                giftCertificate.getPrice() >= 0 && isCertificateConsistsTagsOptionalValid(giftCertificate.getTags());
+    }
+
+    public static boolean isCertificateConsistsTagsOptionalValid(List<Tag> tagList) {
+        if (tagList != null) {
+            for (Tag tag : tagList) {
+                if (!isValidTag(tag)) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    public static boolean moreThenZero(long id) {
+        return id > 0;
+    }
+
+    public static boolean isValidTag(Tag tag) {
+        return tag != null && isStringValid(tag.getName());
+    }
+
+    private static String validateGiftCertificateForUpdating(GiftCertificate giftCertificate) {
+        StringJoiner answer = new StringJoiner(",");
+        if (giftCertificate != null) {
+            if (giftCertificate.getName() != null && StringUtils.isNumeric(giftCertificate.getName())) {
+                answer.add("name " + giftCertificate.getName());
+            }
+            if (!isCertificateConsistsTagsOptionalValid(giftCertificate.getTags())) {
+                answer.add("tags" + giftCertificate.getTags().toString());
+            }
+            return answer.toString();
+        }
+        return "Gift certificate cannot be null for updating";
+    }
+
+    public static Optional<Map<String, String>> isGiftCertificateValidForUpdating(GiftCertificate giftCertificate) {
+
+        Map<String, String> map = new HashMap<>();
+        String answer = validateGiftCertificateForUpdating(giftCertificate);
+
+        if (answer.isEmpty()) {
+            if (giftCertificate.getName() != null) {
+
+                map.put("name", giftCertificate.getName());
+            }
+            if (giftCertificate.getDescription() != null) {
+
+                map.put("description", giftCertificate.getDescription());
+            }
+            if (giftCertificate.getPrice() != null && giftCertificate.getPrice() >= 0) {
+
+                map.put("price", String.valueOf(giftCertificate.getPrice()));
+            }
+            if (giftCertificate.getDuration() != null && giftCertificate.getDuration() >= 0) {
+
+                map.put("duration", String.valueOf(giftCertificate.getDuration()));
+            }
+            if (giftCertificate.getTags() != null && isCertificateConsistsTagsOptionalValid(giftCertificate.getTags())) {
+
+                map.put("tags", "some tags");
+            }
+            return Optional.of(map);
+        }
+        throw new ServerException("Something went wrong, check this fields " + answer);
+    }
+
+    public static boolean isStringValid(String obj) {
+        return obj != null && !StringUtils.isBlank(obj) && !StringUtils.isEmpty(obj) && !StringUtils.isNumeric(obj);
```

Comment:
You don't need to check for null explicitly, StringUtils methods already do it.
You can use StringUtils::isNotBlank, to get rid of inversion.
You don't need to check `isEmpty` after `isBlank`, it already checking this as well.